app/models/user.rb

```diff
@@ -45,19 +45,36 @@ def potential_swap_users(number = 5)
   def create_potential_swaps(number = 5)
     max_attempts = number * 2
     while potential_swaps.reload.count < number
+      try_to_create_marginal_swap
+      max_attempts -= 1
```

Comment:
So this uses up attempts at twice the speed - the question is whether it compensates for that equally by producing swap candidates at twice the speed.  I suspect it's lower than twice because it's presumably much harder to find a marginal swap than a normal potential swap.  So this might result in fewer potential swaps being generated than before, or am I misunderstanding?

Reply:
I think you're right ... I think it makes sense to either double the max_attempts at start or else decrement once in the loop ... probably the former, as it keeps the name 'max_attempts' a bit more honest.

Reply:
@aspiers more elegant solution now on the branch